src/main/webapp/app/pages/curation/CurationPage.tsx

```diff
@@ -194,253 +328,349 @@ const CurationPage = (props: ICurationPageProps) => {
+                        style={{ marginLeft: '0px' }}
+                        type="checkbox"
+                      />
+                      <Label style={{ marginLeft: '18.75px' }}>{filter.label}</Label>
+                    </InputGroup>
+                  </Col>
+                ))}
+              </Row>
+              <h6 className="mb-2 mt-2">Mutation effect</h6>
+              <Row>
+                {mutationEffectFilter.map((filter, index) => (
+                  <Col className="col-6" key={filter.label}>
+                    <InputGroup>
+                      <Input
+                        onChange={() => handleFilterCheckboxChange(index, setMutationEffectFilter)}
+                        checked={filter.selected}
+                        style={{ marginLeft: '0px' }}
+                        type="checkbox"
+                      />
+                      <Label style={{ marginLeft: '18.75px' }}>{filter.label}</Label>
+                    </InputGroup>
+                  </Col>
+                ))}
+              </Row>
+              <h6 className="mb-2 mt-2">Therapeutic implication level of evidence</h6>
+              <Row className="align-items-start justify-content-start">
+                {txLevelFilter.map((filter, index) => (
+                  <Col style={{ flexGrow: 0 }} key={filter.label}>
+                    <InputGroup>
+                      <Input
+                        onChange={() => handleFilterCheckboxChange(index, setTxLevelFilter)}
+                        checked={filter.selected}
+                        style={{ marginLeft: '0px' }}
+                        type="checkbox"
+                      />
+                      <Label style={{ marginLeft: '18.75px' }}>{filter.label}</Label>
+                    </InputGroup>
+                  </Col>
+                ))}
+              </Row>
+              <Row className="justify-content-end">
+                {isFiltered() && (
+                  <Col className="px-0 mr-2" style={{ flexGrow: 0 }}>
+                    <Button
+                      outline
+                      color="danger"
+                      onClick={() => {
+                        setOncogenicityFilter(initFilterCheckboxState(oncogenicityFilterOptions));
+                        setMutationEffectFilter(initFilterCheckboxState(mutationEffectFilterOptions));
+                        setTxLevelFilter(initFilterCheckboxState(txLevelFilterOptions));
+                        setShowFilterModal(false);
+                      }}
+                    >
+                      Reset
+                    </Button>
+                  </Col>
+                )}
+                <Col className="px-0 mr-2" style={{ flexGrow: 0 }}>
+                  <Button
+                    color="primary"
+                    onClick={() => {
+                      handleFilterMutations();
+                      setShowFilterModal(false);
+                    }}
+                  >
+                    Confirm
```

Comment:
Worth to discuss if we want to just have the filter results change as we click on checkboxes or make selections instead of having to make an extra click to apply filters.

Reply:
Discussed with Hongxin. We changed it to a modal, so we the confirm button makes the most sense in this case.